model/opinions.go

```diff
@@ -38,7 +40,61 @@ func newOpinions(lightFile, darkFile string) (*opinions, error) {
 	return result, nil
 }
 
-func (o *opinions) GetOpinionForKey(opinions map[string]interface{}, keyPieces []string) (result interface{}) {
+// FlattenOpinions converts the incoming nested map of opinions into a flat
+// map of properties to values (strings).
+func FlattenOpinions(opinions map[string]interface{}) map[string]string {
+	return flattenOpinionsString(make(map[string]string), "", opinions)
+}
+
+// The following pair of functions differs (only) in the type of the
+// incoming "opinions". The treatment is 99% the same, with the only
+// difference of the Interface variant using Sprintf to convert the
+// key to a string, which it already is for String. The identical
+// parts of both are factored into "flattenOpinionsRecurse".
+//
+// And while we seem to need the String variant only for the toplevel
+// map, as everything below looks to be Interface only, I don't see
+// how to get rid of it either, as a "map[string]interface{}" value
+// cannot be given to a "map[interface{}]interface{}" argument.
+
+func flattenOpinionsString(result map[string]string, prefix string, opinions map[string]interface{}) map[string]string {
+	for ks, value := range opinions {
+		// Here the `ks` iteration variable is already a
+		// string, contrary to flattenOpinionsInterface below.
+
+		result = flattenOpinionsRecurse(result, prefix, ks, value)
+	}
+	return result
+}
+
+func flattenOpinionsInterface(result map[string]string, prefix string, opinions map[interface{}]interface{}) map[string]string {
```

Comment:
Done with commit ec8371468b180046b266e8614f9de8a76273b4f5
(The handling of `cprefix` was a bit more complicated than alluded, mainly because `ks` handling also had to change a bit). Left to do, create the tests for it and the inherited older validation funcss